.circleci/config.yml

```
@@ -0,0 +1,145 @@
+version: 2.1
+
+orbs:
+  aws-cli: circleci/aws-cli@0.1.13
+
+aliases:
+  - &workdir /home/circleci/build
+
+  - &container_config
+    docker:
+      - image: circleci/node:10
+    working_directory: *workdir
+
+  - &checkout checkout
+
+  - &install_aws_cli aws-cli/install
+
+  - &configure_aws_cli
+    aws-cli/configure:
+      aws-region: 'AWS_REGION'
+      aws-access-key-id: 'AWS_ACCESS_KEY_ID'
+      aws-secret-access-key: 'AWS_SECRET_ACCESS_KEY'
+
+  - &save_cache
+    save_cache:
+      paths:
+        - active-chargers-consumer/node_modules
+        - active-chargers-sweep/node_modules
+      key: dependencies-cache-{{ checksum "yarn.lock" }}
+
+  - &restore_cache
+    restore_cache:
+      keys:
+        - dependencies-cache-{{ checksum "yarn.lock" }}
+        # fallback to using the latest cache if no exact match is found
+        - dependencies-cache
+
+  - &install_dependencies
+    run:
+      name: Install Dependencies
+      command: yarn install-all --frozen-lockfile
+
+  - &build_source
+    run:
+      name: Build src into lib
+      command: yarn build-all
+
+  - &run_tests
+    run:
+      name: Run unit tests
+      command: yarn jest
+
+  - &package_stack
+    run:
+      name: Package into packaged-template.yaml
+      command: yarn package
+
+  - &save_to_workspace
+    persist_to_workspace:
+      root: *workdir
+      paths:
+        - packaged-template.yaml
+        - package.json
+        - lib/
+
+  - &mount_workspace
+    attach_workspace:
+      at: *workdir
+
+  - &deploy
+    run:
+      name: Release everything
+      command: yarn deploy
+      no_output_timeout: 30m
+
+  - &feature_branch_filter
+    filters:
+      branches:
+        ignore:
+          - integration
+          - master
+
+  - &production_branch_filter
+    filters:
+      branches:
+        only: master
+
+  - &integration_branch_filter
+    filters:
+      branches:
+        only: integration
+jobs:
```

per @thomashzhu  's comment here: https://voltacharging.atlassian.net/wiki/spaces/SWE/pages/277020891/CircleCI+Configuration+Conventions 

let's change the `jobs` to this:
```suggestion
jobs:
  build-and-test:
    <<: *container_config

    steps:
      - *checkout
      - *configure_npm
      - *install_typescript_globally
      - *restore_cache
      - *install_dependencies
      - *save_cache
      - *build_source
      - *run_tests
      - *remove_local_typescript
      - *save_to_workspace

  deploy:
    <<: *container_config

    steps:
      - *install_aws_cli
      - *configure_aws_cli
      - *mount_workspace
      - *validate_stack
      - *package_stack
      - *deploy_stack
```

---

@begeekandchic Are all of these pieces necessary? I haven't seen some before so I'm not sure what they're meant to do. (Specifically, `configure_npm`, `install_typescript_globally`, and `remove_local_typescript`).

---

`configure_npm` is necessary if this repo uses volta's private npm repositories like @volta/types

`install_typescript_globally` and `remove_local_typescript` is necessary if this repo is written in typescript and needs to be transpiled to a specific version of javascript via `yarn build`.

---

i copied and pasted the above from active-chargers and didn't check if all the steps applied to cars-import. so good catch for asking those questions.

---

Added the TypeScript steps, this stack uses a volta npm package but it's a public one so didn't add the npm step